### PR TITLE
Update PlSqlParser.g4

### PR DESCRIPTION
**Pull request by Pilo11**
- Added `PIPELINE` keyword in function spec rule
- Added `DETERMINISTIC` keyword in `function_body` rule (it does not matter if `DETERMINISTIC` comes first or second after `PIPELINED`)
- Added `RENAME` statement to `unit_statement`
- Added `EDITIONABLE` statement to `create_view` rule
- Re-engineered `WITH` clause (CTE) statement usability for `create_view`, `create_materialzed_view`, `create_table` bodies
- Replaced `'@'` with `AT_SIGN`
- Added `EXISTS`, `PERIOD_KEYWORD` to allowed object names (`regular_ids`)

### CODE REVIEW

File: sql/plsql/PlSqlParser.g4 [hunk 1]
```diff
@@ -176,7 +178,7 @@ procedure_spec
 
 function_spec
     : FUNCTION identifier ('(' parameter ( ',' parameter)* ')')?
-      RETURN type_spec (DETERMINISTIC)? (RESULT_CACHE)? ';'
+      RETURN type_spec (PIPELINED)? (DETERMINISTIC)? (RESULT_CACHE)? ';'
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4]: Please remove excess parentheses:

```g4
RETURN type_spec PIPELINED? DETERMINISTIC? RESULT_CACHE? ';'
```

**Comment by Pilo11** [comment]
> Thanks for your contribution. Is it possible to add pl/sql code samples (see [examples](https://github.com/antlr/grammars-v4/tree/master/sql/plsql/examples) directory) that cover grammar changes?

Of course, sorry. I will add some.

File: sql/plsql/PlSqlParser.g4 (hunk 1 above)
Comment (Pilo11) [reply sql/plsql/PlSqlParser.g4]: Yes, you're right. Unnecessary bloat brackets :D I'll fix that

**Comment by Pilo11** [comment]
Finished requested improvements...

**Comment by KvanTTT** [comment]
Thanks!

**Pull request merged**